roles/srcds/files/addons/sourcemod/scripting/votescramble.sp

```diff
@@ -30,25 +38,26 @@ public void OnPluginStart()
 	cvarVoteChatPercent = CreateConVar("nano_votescramble_chat_percentage", "0.20", "How many players are required for the chat vote to pass? 0.20 = 20%.", 0, true, 0.05, true, 1.0);
 	cvarVoteMenuPercent = CreateConVar("nano_votescramble_menu_percentage", "0.60", "How many players are required for the menu vote to pass? 0.60 = 60%.", 0, true, 0.05, true, 1.0);
 	cvarMinimumVotesNeeded = CreateConVar("nano_votescramble_minimum", "3", "What are the minimum number of votes needed to initiate a chat vote?", 0);
+	cvarSkipSecondVote = CreateConVar("nano_votescramble_skip_second_vote", "0", "Should the second vote be skipped?", 0, true, 0.0, true, 1.0);
 
 	cvarTimeLimit = FindConVar("mp_timelimit");
-	cvarBonusRoundTime = FindConVar("mp_bonusroundtime");
-
-	// HookEvent("teamplay_round_win", Event_RoundEnd);
-	// HookEvent("teamplay_round_stalemate", Event_RoundEnd);
-	// HookEvent("teamplay_win_panel", Event_RoundEnd);
+	cvarMaxRounds = FindConVar("mp_maxrounds");
+	cvarWinLimit = FindConVar("mp_winlimit");
 
 	RegConsoleCmd("sm_votescramble", Cmd_VoteScramble, "Initiate a vote to scramble teams!");
 	RegConsoleCmd("sm_vscramble", Cmd_VoteScramble, "Initiate a vote to scramble teams!");
 	RegConsoleCmd("sm_scramble", Cmd_VoteScramble, "Initiate a vote to scramble teams!");
 	RegAdminCmd("sm_forcescramble", Cmd_ForceScramble, ADMFLAG_VOTE, "Force a team scramble vote.");
+
+	HookEvent("teamplay_round_win", Event_RoundWin);
```

Comment:
I think the correct hook for this is 

```sourcepawn
HookEvent("teamplay_win_panel", Event_RoundEnd);

// later in your Event_RoundEnd function...
if (event.GetInt("round_complete") == 1 || StrEqual(name, "arena_win_panel")) {
    // your logic here... (or invert the if statement to have it be a guard clause)
}
```

This covers certain tf2 spaghetti cases like (I think) arena mode and Hydro.

Reply:
tested this on cp_dustbowl, ctf_2fort, tc_hydro, and arena_badlands... the scramble itself doesn't run on arena_badlands. likely something we'll have to look at in the future.

Reply:
Fair, I think there’s a reason most mods do it this way but it might just as well be inertia.